ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/style/StyleFactory.kt

```diff
@@ -0,0 +1,192 @@
+import com.revenuecat.purchases.paywalls.components.StickyFooterComponent
+import com.revenuecat.purchases.paywalls.components.TextComponent
+import com.revenuecat.purchases.paywalls.components.common.LocalizationDictionary
+import com.revenuecat.purchases.ui.revenuecatui.components.ComponentViewState
+import com.revenuecat.purchases.ui.revenuecatui.components.LocalizedTextPartial
+import com.revenuecat.purchases.ui.revenuecatui.components.PresentedStackPartial
+import com.revenuecat.purchases.ui.revenuecatui.components.ScreenCondition
+import com.revenuecat.purchases.ui.revenuecatui.components.buildPresentedPartial
+import com.revenuecat.purchases.ui.revenuecatui.components.ktx.string
+import com.revenuecat.purchases.ui.revenuecatui.components.ktx.toPaddingValues
+import com.revenuecat.purchases.ui.revenuecatui.components.ktx.toShape
+import com.revenuecat.purchases.ui.revenuecatui.components.properties.toBackgroundStyle
+import com.revenuecat.purchases.ui.revenuecatui.components.properties.toBorderStyle
+import com.revenuecat.purchases.ui.revenuecatui.components.properties.toShadowStyle
+import com.revenuecat.purchases.ui.revenuecatui.components.state.PackageContext
+import com.revenuecat.purchases.ui.revenuecatui.components.toPresentedOverrides
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.VariableDataProvider
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.VariableProcessor
+import com.revenuecat.purchases.ui.revenuecatui.errors.PaywallValidationError
+import com.revenuecat.purchases.ui.revenuecatui.helpers.NonEmptyList
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Result
+import com.revenuecat.purchases.ui.revenuecatui.helpers.map
+import com.revenuecat.purchases.ui.revenuecatui.helpers.mapError
+import com.revenuecat.purchases.ui.revenuecatui.helpers.mapOrAccumulate
+import com.revenuecat.purchases.ui.revenuecatui.helpers.nonEmptyListOf
+import com.revenuecat.purchases.ui.revenuecatui.helpers.orSuccessfullyNull
+import com.revenuecat.purchases.ui.revenuecatui.helpers.zipOrAccumulate
+import java.util.Locale
+
+@Suppress("LongParameterList")
+internal class StyleFactory(
+    private val windowSize: ScreenCondition,
+    private val isEligibleForIntroOffer: Boolean,
+    private val componentState: ComponentViewState,
+    private val packageContext: PackageContext,
+    private val localizationDictionary: LocalizationDictionary,
+    private val locale: Locale,
+    private val variables: VariableDataProvider,
+) {
+
+    private companion object {
+        private const val DEFAULT_SPACING = 0f
+        private val DEFAULT_SHAPE = RectangleShape
+    }
+
+    @Composable
+    fun create(component: PaywallComponent): Result<ComponentStyle, NonEmptyList<PaywallValidationError>> =
+        when (component) {
+            is ButtonComponent -> TODO("ButtonComponentStyle is not yet implemented.")
+            is ImageComponent -> TODO("ImageComponentStyle is not yet implemented.")
```

Comment:
This is not entirely true anymore, but I'd like to verify that this approach works before we add support for all the other `ComponentStyle`s.